generate_if_map.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define GATEKEEPER_IF_MAP	"./lua/if_map.lua"
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <ifaddrs.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+#include <linux/ethtool.h> 
+#include <linux/sockios.h>
+
+int
+main(void)
+{
+	FILE *f;
+	struct ifaddrs *addrs, *iter;
+	int sock;
+	int ret;
+
+	f = fopen(GATEKEEPER_IF_MAP, "w");
+	if (f == NULL) {
+		perror("fopen");
+		return -1;
+	}
+
+	ret = getifaddrs(&addrs);
+	if (ret == -1) {
+		perror("getifaddrs");
+		goto file;
+	}
+
+	sock = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sock == -1) {
+		perror("socket");
+		goto addrs;
+	}
+
+	fprintf(f, "return {\n");
+
+	iter = addrs;
+	while (iter != NULL) {
+		struct ifreq ifr;
+		struct ethtool_cmd cmd;
+		struct ethtool_drvinfo drvinfo;
+
+		/*
+		 * Use AF_PACKET to only get each interface once,
+		 * and skip the loopback interface.
+		 */
+		if (iter->ifa_addr == NULL ||
+				iter->ifa_addr->sa_family != AF_PACKET ||
+				strncmp(iter->ifa_name, "lo", 2) == 0)
```

Comment:
What's the intention here: (1) skipping the interface "lo", or (2) skipping all interfaces whose names start with "lo"? The current code is implementing the second case, but I think that the intention is the first one.